library/modules/bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,836 @@
+            return ''
+        return self._values['description']
+
+    @property
+    def address(self):
+        if self._values['address'] is None:
+            return None
+        return self._values['address']
+
+    @property
+    def port(self):
+        if self._values['port'] is None:
+            return None
+        return self._values['port']
+
+    @property
+    def ip_protocol(self):
+        if self._values['ip_protocol'] is None:
+            return None
+        return self._values['ip_protocol']
+
+    @property
+    def mask(self):
+        if self._values['mask'] is None:
+            return None
+        return self._values['mask']
+
+    @property
+    def translate_address(self):
+        if self._values['translate_address'] is None:
+            return None
+        return self._values['translate_address']
+
+    @property
+    def translate_port(self):
+        if self._values['translate_port'] is None:
+            return None
+        return self._values['translate_port']
+
+    @property
+    def advertise(self):
+        if self._values['advertise'] is None:
+            return None
+        return flatten_boolean(self._values['advertise'])
+
+    @property
+    def vlans_enabled(self):
+        if self._values['vlans_enabled'] is None:
+            return None
+        elif self._values['vlans_enabled'] is True:
+            return True
+        return False
+
+    @property
+    def vlans_disabled(self):
+        if self._values['vlans_disabled'] is None:
+            return None
+        elif self._values['vlans_disabled'] is True:
+            return True
+        return False
+
+    @property
+    def vlans(self):
+        if self._values['vlans'] is None:
+            return None
+        if is_empty_list(self._values['vlans']):
+            return []
+        result = list()
+        for item in self._values['vlans']:
+            result.append(item)
+        return result
+
+    @property
+    def rules(self):
+        if self._values['rules'] is None:
+            return None
+        if is_empty_list(self._values['rules']):
+            return ''
+        results = list()
+        for irule in self._values['rules']:
+            result = fq_name(self.partition, irule)
+            results.append(result)
+        return results
+
+    @property
+    def last_hop_pool(self):
+        if self._values['last_hop_pool'] is None:
+            return None
+        return self._values['last_hop_pool']
+
+    @property
+    def source_port(self):
+        if self._values['source_port'] is None:
+            return None
+        return self._values['source_port']
+
+    @property
+    def fallback_persistence(self):
+        if self._values['fallback_persistence'] is None:
+            return None
+        return self._values['fallback_persistence']
+
+    @property
+    def auto_lasthop(self):
+        if self._values['auto_lasthop'] is None:
+            return None
+        return self._values['auto_lasthop']
```

Comment:
why are those here?

Reply:
I agree - since no transformation is done on those properties, defining them here is unnecessary. 

This comment is addressed now